docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md

```diff
@@ -78,15 +78,248 @@ If you need to have external labels in your query results, this is currently not
+   ```
+   
+   
+   Cleanup the original blocks which are marked for deletion.
+  
+   ```bash
+   thanos tools bucket cleanup \
+       --objstore.config-file bucket.yaml \
+       --delete-delay=0
+   ```
+  
+   > **Note**: If there are multiple prometheus clusters, then relabelling each of them in parallel may speed up the entire process.
+   > Get the list of blocks that for each cluster, and process it separately.
+   > ```bash
+   > thanos tools bucket inspect \
+   >     --objstore.config-file bucket.yaml \
+   >     --output=tsv \
+   >     | grep <PROMETHEUS-CLUSTER-NAME> \
+   >     | awk '{print $1}' > prod-blocks.tsv
+   >```
+  
+   >```bash
+   > for ID in `cat prod-blocks.tsv`
+   > do
+   >     thanos tools bucket rewrite  \
+   >        --objstore.config-file bucket.yaml \
+   >        --rewrite.to-relabel-config-file relabel-config.yaml \
+   >        --delete-blocks \
+   >        --no-dry-run \
+   >        --id $ID         
+   > done
+   >```
+ 
+5) Remove external labels from meta.json
+   
+   Mimir compactor will not be able to compact the blocks having external labels with Mimir's own blocks that don't have any such labels in their meta.json. Therefore these external labels have to be removed before copying them to the Mimir bucket.
+   
+   Use the below script to remove the labels from the meta.json.
+   
+   ```bash
+   #!/bin/bash
+
+   BUCKET="GCS Bucket name"
+
+   echo "Fetching list of meta.json files (this can take a while if there are many blocks)"
+   gsutil ls "gs://$BUCKET/*/meta.json" > meta-files.txt
+
+   echo "Processing meta.json files"
+   for FILE in $(cat meta-files.txt); do
+      echo "Removing Thanos labels from $FILE"
+      ORIG_META_JSON=$(gsutil cat "$FILE")
+      UPDATED_META_JSON=$(echo "$ORIG_META_JSON" | jq "del(.thanos.labels)")
+
+      if ! diff -u <( echo "$ORIG_META_JSON" | jq . ) <( echo "$UPDATED_META_JSON" | jq .) > /dev/null; then
+         echo "Backing up $FILE to $FILE.orig"
+         gsutil cp "$FILE" "$FILE.orig"
+         echo "Uploading modified $FILE"
+         echo "$UPDATED_META_JSON" | gsutil cp - "$FILE"
+      else
+         echo "No diff for $FILE"
+      fi
+   done
+   ```
+   
+6) Copy the blocks from the intermediate bucket to the Mimir bucket
+  
+   Copy the blocks from the intermediate bucket to the Mimir bucket.
+  
+   ```bash
+   gsutil -m cp -r gs://<mimir-intermediate-bucket> gs://<mimir-gcs-bucket>/<TENANT>/
+   ```
+ 
+7) Mimir compactor updates bucket index
+  
+   Historical blocks will not be available for querying as soon as they are uploaded, because the bucket index with the list of all available blocks needs to be updated by the compactor first. Compactor typically does this update every 15 minutes. Once this is done, other components like querier or store-gateway will be able to work with these the historical blocks, and blocks will be available for querying through Grafana. 
```

Comment:
```suggestion
   Historical blocks are not available for querying immediately after they are uploaded because the bucket index with the list of all available blocks first needs to be updated by the compactor. The compactor typically perform such an update every 15 minutes. After an update completes, other components such as the querier or store-gateway are able to work with the historical blocks, and the blocks are available for querying through Grafana. 
```